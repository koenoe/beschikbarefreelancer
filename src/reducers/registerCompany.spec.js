// @flow
import registerCompanyReducer from 'reducers/registerCompany';

import type { registerCompanyState } from 'reducers/registerCompany';

describe('Reducer: registerCompany', () => {
  it('should set `isLoading` to true when `REGISTER_COMPANY_REQUEST`', () => {
    const beforeState: registerCompanyState = {
      isLoading: false,
    };
    const afterState: registerCompanyState = registerCompanyReducer(beforeState, {
      type: 'REGISTER_COMPANY_REQUEST',
      email: 'test@koenromers.com',
    });
    expect(afterState).toMatchSnapshot();
    expect(afterState).not.toBe(beforeState);
  });

  it('should set `success` to true when `REGISTER_COMPANY_SUCCESS`', () => {
    const beforeState: registerCompanyState = {
      isLoading: true,
      success: false,
    };
    const afterState: registerCompanyState = registerCompanyReducer(beforeState, {
      type: 'REGISTER_COMPANY_SUCCESS',
      success: true,
    });
    expect(afterState).toMatchSnapshot();
    expect(afterState).not.toBe(beforeState);
  });

  it('should have an `error` when `REGISTER_COMPANY_FAILURE`', () => {
    const beforeState: registerCompanyState = {
      isLoading: true,
      error: null,
    };
    const afterState: registerCompanyState = registerCompanyReducer(beforeState, {
      type: 'REGISTER_COMPANY_FAILURE',
      error: new Error('fail whale'),
    });
    expect(afterState).toMatchSnapshot();
    expect(afterState).not.toBe(beforeState);
  });
});
