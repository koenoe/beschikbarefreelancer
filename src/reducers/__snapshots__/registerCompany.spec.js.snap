// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: registerCompany should have an \`error\` when \`REGISTER_COMPANY_FAILURE\` 1`] = `
Object {
  "error": [Error: fail whale],
  "isLoading": false,
}
`;

exports[`Reducer: registerCompany should set \`isLoading\` to true when \`REGISTER_COMPANY_REQUEST\` 1`] = `
Object {
  "isLoading": true,
}
`;

exports[`Reducer: registerCompany should set \`success\` to true when \`REGISTER_COMPANY_SUCCESS\` 1`] = `
Object {
  "isLoading": false,
  "success": true,
}
`;
